documentation/src/main/asciidoc/userguide/chapters/fetching/Fetching.adoc

```diff
@@ -248,6 +251,144 @@ include::{extrasdir}/fetching-strategies-dynamic-fetching-entity-subgraph-exampl
+----
+====
+
+This example actually functions exactly as <<fetching-strategies-dynamic-fetching-entity-subgraph-example>>,
+just using a parsed graph rather than a named graph.
+
+
+The syntax also supports defining "key sub-graphs".  To specify a key sub-graph, `.key` is added
+to the end of the attribute name.
+
+.Parsing an entity key graph
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/GraphParsingTest.java[tags=fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-1]
+----
+====
+
+.Parsing a map key graph
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/GraphParsingTest.java[tags=fetching-strategies-dynamic-fetching-entity-graph-parsing-key-example-2]
+----
+====
+
+Parsing can also handle sub-type specific sub-graphs.  For example, given an entity hierarchy of
+`LegalEntity` <- (`Corporation` | `Person` | `NonProfit`) and an attribute named `responsibleParty` whose
+type is the `LegalEntity` base type we might have:
+
+====
+[source, JAVA, indent=0]
+----
+responsibleParty(Corporation: ceo)
+----
+====
+
+We can even duplicate the attribute names to apply different sub-type sub-graphs:
+
+====
+[source, JAVA, indent=0]
+----
+responsibleParty(taxIdNumber), responsibleParty(Corporation: ceo), responsibleParty(NonProfit: sector)
+----
+====
+
+The duplicated attribute names are handled according to the JPA specification which says that duplicate
+specification of the attribute node results in the originally registered AttributeNode to be re-used
+effectively merging the 2 AttributeNode specifications together.  In other words, the above specification
+creates a single AttributeNode with 3 distinct SubGraphs.  It is functionally the same as calling:
+
+====
+[source, JAVA, indent=0]
+----
+Class<Invoice> invoiceClass = ...;
+javax.persistence.EntityGraph<Invoice> invoiceGraph = entityManager.createEntityGraph( invoiceClass );
+invoiceGraph.addAttributeNode( "responsibleParty" );
+invoiceGraph.addSubgraph( "responsibleParty" ).addAttributeNode( "taxIdNumber" );
+invoiceGraph.addSubgraph( "responsibleParty", Corporation.class ).addAttributeNode( "ceo" );
+invoiceGraph.addSubgraph( "responsibleParty", NonProfit.class ).addAttributeNode( "sector" );
+----
+====
+
+[[fetching-strategies-dynamic-fetching-entity-graph-merging]]
+==== Combining multiple JPA entity graphs into one
+
+Multiple entity graphs can be combined into a single supergraph that acts as a union. Graph from the previous example can also be build by combining separate aspect graphs into one, such as:
+
+.Combining multiple graphs into one
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/GraphParsingTest.java[tags=fetching-strategies-dynamic-fetching-entity-graph-merging-example]
+----
+====
+
+[[fetching-strategies-dynamic-fetching-entity-graph-applying]]
+==== Applying entity graphs to queries
+
+Convenience/utility methods were created to make application of entity graphs to queries easier. Following examples show
```

Comment:
Not sure the sentence is finished?

Reply:
Probably.  I'll have to look

Reply:
Actually I think this whole section (`fetching-strategies-dynamic-fetching-entity-graph-applying`) should just go away.  It was originally trying to discuss the proprietary methods like `org.hibernate.graph.EntityGraphs#executeList`.  Initially there were many many more such methods.  TBH I personally think even these executeList ones should maybe go away...

The previous sections that @vladmihalcea had already written cover how to apply graphs to query and load/find operations.